### PR TITLE
fix(linux): Gracefully handle makedirs failures

### PR DESCRIPTION
**Pull request by ermshiperete**
This change catches the exception we're getting if we try to create a directory but e.g. there is already a file with that name. In this case we now output an error message. This change also gracefully handles failures when linking files.

Fixes #6285.

# User Testing

## Preparations

- The Linux version doesn't matter for these tests.
- Install build artifacts of this PR
- Simulate the corrupt environment by running:
  ```bash
  mv ~/.local/share/fonts ~/.local/share/fonts.bak
  touch ~/.local/share/fonts
  ```

After running the tests if you don't restore a snapshot, you might want to run the following command to get things back into a working state:
```bash
rm ~/.local/share/fonts
mv ~/.local/share/fonts.bak ~/.local/share/fonts
```

## Tests

**TEST_KB_INSTALL**: Install a keyboard

- On the command line run `km-config`, click "Download", enter search for "Khmer", and install the "Khmer Angkor" keyboard
- Verify that the "Khmer Angkor" keyboard got installed and appears in the list of keyboards in the "Keyman Configuration" dialog.
- Verify that the terminal window shows 4 lines that start with `ERROR:Can't create directory /home/...`

### CODE REVIEW

File: linux/keyman-config/keyman_config/install_kmp.py [hunk 1]
```diff
@@ -101,8 +101,8 @@ def _install_kmp(self, inputfile, online, language, area):
         self.kmpdocdir = get_keyman_doc_dir(area, self.packageID)
         self.kmpfontdir = get_keyman_font_dir(area, self.packageID)
 
-        if not os.path.isdir(self.packageDir):
-            os.makedirs(self.packageDir)
+        if not self._safeMakeDirs(self.packageDir):
```
Comment (darcywong00) [inline linux/keyman-config/keyman_config/install_kmp.py:104]: This used to check a boolean. Is the same as 
`if self._safeMakeDirs(self.packageDir) is None`?

**Review by mcdurdin — APPROVED** [review]

**Comment by bharanidharanj** [comment]
* **TEST_KB_INSTALL (PASSED):** Tested this issue in Impish 20.04 Linux OS. The output line shows an error message after installing the Keyboard.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.206-beta](https://keyman.com/downloads/releases/beta/15.0.206)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.205-alpha](https://keyman.com/downloads/releases/alpha/15.0.205)

**Comment by ermshiperete** [comment]
Oops, got the issue# wrong. This actually fixes #6286.

**Pull request merged**